tracim/tests/library/test_user_api.py

```diff
@@ -60,26 +63,84 @@ def test_get_one_by_email_exception(self):
+    #         current_user=None,
+    #         session=self.session,
+    #         config=self.config,
+    #     )
+    #     u1 = api.create_user(True)
+    #     u2 = api.create_user(True)
+    #     users = api.get_all()
+    #     assert 2==len(users)
+
+    def test_unit__get_one__ok__nominal_case(self):
         api = UserApi(
             current_user=None,
             session=self.session,
             config=self.config,
         )
-        # u1 = api.create_user(True)
-        # u2 = api.create_user(True)
+        u = api.create_user()
+        api.update(u, 'titi', 'titi@titi', True)
+        one = api.get_one(u.user_id)
+        eq_(u.user_id, one.user_id)
 
-        # users = api.get_all()
-        # ok_(2==len(users))
+    def test_unit__get_user_with_context__nominal_case(self):
+        user = User(email='admin@tracim.tracim')
+        api = UserApi(
+            current_user=None,
+            session=self.session,
+            config=self.config,
+        )
+        new_user = api.get_user_with_context(user)
+        assert isinstance(new_user, UserInContext)
+        assert new_user.user == user
 
-    def test_get_one(self):
+    def test_unit__get_current_user_ok__nominal_case(self):
+        user = User(email='admin@tracim.tracim')
+        api = UserApi(
+            current_user=user,
+            session=self.session,
+            config=self.config,
+        )
+        new_user = api.get_current_user()
+        assert isinstance(new_user, User)
+        assert user == new_user
+
+    def test_unit__get_current_user__err__user_not_exist(self):
         api = UserApi(
             current_user=None,
             session=self.session,
             config=self.config,
         )
-        u = api.create_user()
-        api.update(u, 'titi', 'titi@titi', True)
-        one = api.get_one(u.user_id)
-        eq_(u.user_id, one.user_id)
+        with pytest.raises(UserNotExist):
+            api.get_current_user()
+
+    def test_unit__authenticate_user___ok__nominal_case(self):
+        api = UserApi(
+            current_user=None,
+            session=self.session,
+            config=self.config,
+        )
+        user = api.authenticate_user('admin@admin.admin', 'admin@admin.admin')
+        assert isinstance(user, User)
+        assert user.email == 'admin@admin.admin'
+
+    def test_unit__authenticate_user___err__bad_password(self):
+        api = UserApi(
+            current_user=None,
+            session=self.session,
+            config=self.config,
+        )
+        with pytest.raises(AuthenticationFailed):
+            api.authenticate_user('admin@admin.admin', 'bad_password')
+
+    def test_unit__authenticate_user___err__bad_user(self):
+        api = UserApi(
+            current_user=None,
+            session=self.session,
+            config=self.config,
+        )
+        with pytest.raises(AuthenticationFailed):
+            api.authenticate_user('unknown_user', 'bad_password')
```

Comment:
bad password means "un mot de passe de mauvaise qualité". wrong password means "un mot de passe erroné". you should use WrongPassword (even rename the exception BadPassword to WrongPassword)